packages/client3/components/design-system/button/button.tsx

```diff
@@ -1,163 +1,93 @@
-          : colorScheme === "sky"
-          ? "bg-sky-800 text-white hover:bg-sky-900 active:bg-sky-900"
-          : null,
-        className
-      )}
-      disabled={_disabled}
-      type={type}
-      {...rest}
-    >
-      {spinnerOnLeft ? (
-        <Spinner size="sm" className={clsx(children ? "-my-2 -ml-1" : null)} />
-      ) : iconLeft ? (
-        <Icon
-          name={iconLeft}
-          size="sm"
-          className={clsx(children ? "-my-2 -ml-1" : null)}
-        />
-      ) : null}
-      {children}
-      {spinnerOnRight ? (
-        <Spinner size="sm" className={clsx(children ? "-my-2 -mr-1" : null)} />
-      ) : iconRight ? (
-        <Icon
-          name={iconRight}
-          size="sm"
-          className={clsx(children ? "-my-2 -mr-1" : null)}
-        />
-      ) : null}
-    </Component>
-  );
-});
-
-interface IconButtonProps
-  extends Omit<ButtonProps, "children" | "iconLeft" | "iconRight"> {
-  icon: IconProps["name"];
+export type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
   /**
-   * Accessibility label. Must be provided since this form of button has no visible label.
+   * Content within the button
    */
-  label: string;
-}
+  children?: ReactNode;
+  disabled?: boolean;
+  buttonType?: "primary" | "secondary" | "tertiary";
+  iconLeft?: IconProps["name"];
+  iconRight?: IconProps["name"];
+  isLoading?: boolean;
+};
 
-export const IconButton = forwardRef<HTMLButtonElement, IconButtonProps>(
-  function IconButton({ icon, size = "md", className, ...rest }, ref) {
+export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+  function Button(
+    {
+      children,
+      iconLeft,
+      iconRight,
+      disabled,
+      type,
+      buttonType = "primary",
+      isLoading = false,
+      className,
+      ...rest
+    },
+    ref
+  ) {
+    const formContext = useFormContext();
+    let _disabled = disabled;
+    let _isLoading = isLoading;
+    if (formContext !== null && type === "submit") {
+      const {
+        formState: { isSubmitting, errors },
+      } = formContext;
+      // Can't use the built-in isValid on formState because that only updates when the mode is set to onChange/onBlur/whatever
+      const isValid = Object.keys(errors).length === 0;
+      _disabled = disabled || isSubmitting || !isValid;
+      _isLoading = isSubmitting;
+    }
+    const spinnerOnLeft = _isLoading && iconLeft;
+    const spinnerOnRight = _isLoading && !spinnerOnLeft;
     return (
-      <Button
-        iconRight={icon}
+      <button
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        ref={ref as any}
         className={clsx(
-          size === "sm"
-            ? "!p-0.5"
-            : size === "md"
-            ? "!p-1.5"
-            : size === "lg"
-            ? "!p-2.5"
-            : size === "xl"
-            ? "!p-[1.125rem]"
-            : null,
+          "rounded-[100px] ",
```

Comment:
Would `rounded-full` work? 

Ref: https://tailwindcss.com/docs/border-radius#pill-buttons

Reply:
Yeah, I initially used `rounded-full` which resolves to `border-radius: 9999px`then changed it as in designs the required border radius was 100px. Saying that visually I couldn't find any differences in both the CSS styles.  Which do you think is better?